user_tools/src/spark_rapids_pytools/common/utilities.py

```diff
@@ -233,16 +248,31 @@ def get_and_setup_logger(cls, type_label: str, debug_mode: bool = False):
         cls._ensure_configured(debug_enabled)
 
         logger = logging.getLogger(type_label)
-        log_file = Utils.get_rapids_tools_env('LOG_FILE')
-        # Ensure multiple handlers are not added
-        if log_file and not logger.handlers:
-            fh = logging.FileHandler(log_file)
-            fh.setLevel(logging.DEBUG)
-            formatter = logging.Formatter('{asctime} {levelname} {name}: {message}', style='{')
-            fh.setFormatter(formatter)
-            logger.addHandler(fh)
+
+        cls._rebind_file_handler(logger, Utils.get_rapids_tools_env('LOG_FILE'))
         return logger
 
+    @classmethod
+    def _rebind_file_handler(cls, logger: logging.Logger, log_file: str) -> None:
+        # Remove existing FileHandlers to avoid stale paths and duplicates
```

Comment:
just for clarification for someone who comes later and reads this. Can we just give a brief of what is meant by stale/duplicate paths and what scenarios cause them?